packages/dds/tree/src/test/forestTestSuite.ts

```diff
@@ -0,0 +1,724 @@
+                assert.equal(reader.value, 4);
+                assert.equal(reader.nextNode(), false);
+                reader.exitField();
+                assert.equal(reader.nextNode(), true);
+                assert.equal(reader.value, 1);
+                assert.equal(reader.nextNode(), true);
+                assert.equal(reader.value, 2);
+                assert.equal(reader.nextNode(), false);
+            });
+
+            it("modify and delete", () => {
+                const forest = factory(new InMemoryStoredSchemaRepository(defaultSchemaPolicy));
+                initializeForest(forest, nestedContent.map(singleTextCursor));
+
+                const mark: Delta.ModifyAndDelete = {
+                    type: Delta.MarkType.ModifyAndDelete,
+                    fields: new Map([[xField, []]]),
+                };
+                const delta: Delta.Root = new Map([[rootFieldKeySymbol, [mark]]]);
+                // TODO: make type-safe
+                forest.applyDelta(delta);
+
+                const reader = forest.allocateCursor();
+                moveToDetachedField(forest, reader);
+                assert.equal(reader.firstNode(), false);
+                reader.enterField(xField);
+                assert.equal(reader.firstNode(), false);
+            });
+
+            it("modify and move out", () => {
+                const forest = factory(new InMemoryStoredSchemaRepository(defaultSchemaPolicy));
+                initializeForest(forest, nestedContent.map(singleTextCursor));
+
+                const moveId = brandOpaque<Delta.MoveId>(0);
+                const mark: Delta.Modify = {
+                    type: Delta.MarkType.Modify,
+                    fields: new Map([
+                        [
+                            xField,
+                            [
+                                {
+                                    type: Delta.MarkType.ModifyAndMoveOut,
+                                    setValue: 2,
+                                    moveId,
+                                },
+                            ],
+                        ],
+                        [yField, [{ type: Delta.MarkType.MoveIn, moveId }]],
+                    ]),
+                };
+                const delta: Delta.Root = new Map([[rootFieldKeySymbol, [mark]]]);
+                // TODO: make type-safe
+                forest.applyDelta(delta);
+
+                const reader = forest.allocateCursor();
+                moveToDetachedField(forest, reader);
+                assert(reader.firstNode());
+                reader.enterField(xField);
+                assert.equal(reader.getFieldLength(), 0);
+                reader.exitField();
+                reader.enterField(yField);
+                assert(reader.firstNode());
+                assert.equal(reader.value, 2);
+                assert.equal(reader.nextNode(), true);
+            });
+
+            // TODO:#2592: Unskip once MoveInAndModify is properly supported.
```

Comment:
I don't think we've adopted the `TODO:#<AdoID>` convention because the code is open source and external contributors don't have access to ADO work items. We haven't adopted the convention to *not* do it either AFAIK. Maybe someting to bring up for discussion at standup or office hours.

Reply:
Oh that's a good point. I think not having it makes sense, but it would then also be nice for us to have a way to link them. Maybe just have the title of the item somewhere so it's searchable?